File: Desktop/SummerVacation/0704/src/numThirtySeven.java
import java.util.*;

public class numThirtySeven {
    public static void main(String[] args) {
        Random makePassword = new Random();
        Scanner scan = new Scanner(System.in);
        int length,numbers,characters,word;
        List<Object>password = new ArrayList();
        System.out.print("What's the minimum length? ");
        length = scan.nextInt();
        System.out.print("How many special characters? ");
        characters = scan.nextInt();
        System.out.print("How many numbers? ");
        numbers = scan.nextInt();
        //97~122 char, 33~38 special
        for(int count = 0 ; count < characters; count++) {
                word = makePassword.nextInt(15) + 33;
                password.add((char)word);
        }
        for(int count = 0; count < numbers; count++){
                word = makePassword.nextInt(10);
                password.add(word);
        }
        if(length > characters+numbers){
            for(int count = 0; count < length - characters - numbers ; count++){
                word = makePassword.nextInt(26)+ 97;
                password.add((char)word);
            }
        }
        else{
            for(int count = 0; count < makePassword.nextInt(5)+1;count++){
                word = makePassword.nextInt(26) + 97;
                password.add((char)word);
            }
        }
        System.out.print("Your password is ");
        for(Object alpha : password){
            System.out.print(alpha);
        }
    }
}

File: Desktop/SummerVacation/0704/python/num_thirty_seven.py
import random
length = input("What's the minimum length? ")
special = input("How many special characters? ")
numbers = input("How many numbers? ")
special_word = ['!','@','#','$','%','^','&','*','~']
password = '\0'
length = int(length) - int(special) - int(numbers);
for count in range(0,int(special)) :
    password = password + random.choice(special_word)
for count in range(0,int(numbers)) :
    password = password + str(random.randrange(0,10))
if length < 1 :
    for count in range(0,random.randrange(1,6)) :
        password = password + chr(random.randrange(97,123))
else :
    for count in range(0,length)  :
        password = password + chr(random.randrange(97,123))
print(password)